### ISSUE TITLE
BERT作法

### ISSUE THREAD

**Issue by WenTingTseng** [p208p2002/taipei-QA-BERT#8]
請問您的做法是整個BERT+FFN下去Fine Tune訓練，還是BERT的部分固定參數只有後面FFN訓練呢?

謝謝

**Comment by p208p2002** [comment]
全部下去一起訓練哦
根據我的經驗，大部分情況下凍結BERT只訓練FFN會產生比較差的結果

**Comment by WenTingTseng** [comment]
喔喔 了解，但是我是有聽說資料量比較少的情況怕會把BERT Fine tune壞掉，所以先凍結BERT只訓練FFN。
那如果在這範例中我要凍結BERT的話應該如何修改呢

**Comment by p208p2002** [comment]
方法大致上有兩種
一種是設定layer gradient = False
另外一種是設定 optimizer 將該layer 的lr調為0

可以參閱
https://discuss.pytorch.org/t/how-the-pytorch-freeze-network-in-some-layers-only-the-rest-of-the-training/7088/11

當不知道layer name的時候推薦把model印出來看
> print(model)

**Comment by WenTingTseng** [comment]
ok，了解，謝謝你
